pkg/koko/koko.go

```diff
@@ -55,9 +56,31 @@ func RunForever(confPath string) {
 func bootstrap() {
 	i18n.Initial()
 	logger.Initial()
+}
+
+func bootstrapWithJMService(jmsService *service.JMService) {
+	updateEncryptConfigValue(jmsService)
 	exchange.Initial()
 }
 
+func updateEncryptConfigValue(jmsService *service.JMService) {
+	cfg := config.GlobalConfig
+	encryptKey := cfg.SecretEncryptKey
+	if encryptKey != "" {
+		redisPassword := cfg.RedisPassword
+		ret, err := jmsService.GetEncryptedConfigValue(encryptKey, redisPassword)
+		if err != nil {
+			logger.Error("Get encrypted config value failed: " + err.Error())
+			return
+		}
+		if ret.Value != "" {
+			cfg.UpdateRedisPassword(ret.Value)
+		} else {
+			logger.Error("Get encrypted config value failed: empty value")
+		}
+	}
+}
+
 func runTasks(jmsService *service.JMService) {
 	if config.GetConf().UploadFailedReplay {
 		go uploadRemainReplay(jmsService)
```

Comment:
There's no mention of 'uploadRemainReplay' function in your code. The `uploadRemainReplay` function is defined somewhere within the project but not shown here.

Additionally, to optimize further:

1. Consider using a channel instead if you expect only one signal at a time when notifying shutdown.
1. Avoid creating unnecessary channels (`ch := make(chan ...)`, etc., which may be inefficient). Instead, consider making variables local or sharing them across multiple functions/variables where appropriate.

I'll suggest checking these parts with the actual implementation so I can provide more relevant information from there on out.